backend/manager.go

```diff
@@ -73,7 +73,7 @@ func (bm *manager) GetBackend(backendType string) (Backend, error) {
 
 	bm.wg.Add(1)
 	go func() {
-		be.Run(bm.ctx)
+		<-bm.ctx.Done()
```

Comment:
Thinking out loud here: shouldn't we be able to just remove `ctx` field from `manager` struct? Seems like we're not using it anywhere, just closing it here.